src/components/textAreas/textAreaRichText/textAreaRichTextActions.tsx

```diff
@@ -0,0 +1,95 @@
+import type { Editor } from '@tiptap/react';
+import classNames from 'classnames';
+import { useCallback, useMemo } from 'react';
+import { Button } from '../../button';
+import { IconType } from '../../icon';
+
+export interface ITextAreaRichTextAction {
+    /**
+     * Icon of the action.
+     */
+    icon: IconType;
+    /**
+     * TipTap command or callback called on action click.
+     */
+    action?: () => unknown;
+    /**
+     * Hides the action when set to true.
+     */
+    hidden?: boolean;
+}
+
+export interface ITextAreaRichTextActionsProps {
+    /**
+     * Instance of the TipTap editor.
+     */
+    editor: Editor | null;
+    /**
+     * Renders the actions as disabled when set to true.
+     */
+    isDisabled?: boolean;
+    /**
+     * Callback called on expand action click.
+     */
+    onExpandClick?: () => void;
+}
+
+export const TextAreaRichTextActions: React.FC<ITextAreaRichTextActionsProps> = (props) => {
+    const { editor, isDisabled, onExpandClick } = props;
+
+    const unsetLink = useCallback(() => editor?.chain().focus().extendMarkRange('link').unsetLink().run(), [editor]);
+
+    const setLink = useCallback(() => {
+        const previousUrl = editor?.getAttributes('link').href;
+        const url = window.prompt('URL', previousUrl);
+
+        if (url == null || url === '') {
+            unsetLink();
+
+            return;
+        }
+
+        editor?.chain().focus().extendMarkRange('link').setLink({ href: url }).run();
+    }, [editor, unsetLink]);
+
+    const richTextActions = useMemo(() => {
```

Comment:
Good catch, fixed the set/unset handling. Regarding exiting the "link" mode without any content after the link, I'm not sure if this is intended by the library so that you can add the url and then insert the name of the link. In any case the current component on the App behaves the same, I would suggest to further investigate on this if we get such a feedback from design :)